drivers/mfd/mfd_nct38xx.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 Google, LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_nct38xx
+
+#include <zephyr/drivers/i2c.h>
+
+struct mfd_nct38xx_config {
+	const struct i2c_dt_spec i2c_dev;
+};
+
+struct mfd_nct38xx_data {
+	/* lock NC38xx register access */
+	struct k_sem lock;
+};
+
+static int mfd_nct38xx_init(const struct device *dev)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	k_sem_init(&data->lock, 1, 1);
+
+	return 0;
+}
+
+int mfd_nct38xx_lock(const struct device *dev, k_timeout_t timeout)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	return k_sem_take(&data->lock, timeout);
+}
+
+void mfd_nct38xx_unlock(const struct device *dev)
+{
+	struct mfd_nct38xx_data *data = dev->data;
+
+	k_sem_give(&data->lock);
+}
```

Comment:
this seems like just a wrapper around  `k_sem_take/give`. Wouldn't it be better to allow getting a reference to a shared lock..? So drivers don't need to use a new idiom for locking.

Reply:
Changed to return a reference the semaphore.